octodns/manager.py

```diff
@@ -303,7 +341,7 @@ def configured_sub_zones(self, zone_name):
         if self._configured_sub_zones is None:
             # First time through we compute all the sub-zones
 
-            configured_sub_zones = {}
+            configured_sub_zones = IdnaDict()
```

Comment:
This allows us to ask for sub-zones with either utf-8 names or idna encoded. What we get back will always be the idna encoded subs. I'm adding a comment to that effect now.